File: python/0977有序数组的平方.py
nums = [1]


def sortedSquares1(nums:list):
    
    return sorted([i*i for i in nums])

def sortedSquares2(nums:list):
    l=0
    r=len(nums)-1
    res=[]
    while(l<=r):
        if(abs(nums[l])<abs(nums[r])):
            res.append(nums[r]*nums[r])
            r-=1
        else:
            res.append(nums[l]*nums[l])
            l+=1
    
    return res[::-1]

print(sortedSquares2(nums))

File: cpp/0977有序数组的平方.cpp
#include<vector>
#include<iostream>
#include<math.h>

using namespace std;

vector<int> sortedSquares(vector<int> &nums)
{
    int left = 0;
    int s = nums.size();
    int right = s-1;
    vector<int> res(s);
    --s;
    while (left <= right)
    {
        if(abs(nums[left])>abs(nums[right]))
        {
            res[s--] = nums[left] * nums[left];
            ++left;
        }
        else
        {
            res[s--] = nums[right] * nums[right];
            --right;
        }
    }

    return res;
}

int main()
{
    vector<int> nums = {1, 2, 3, 4};

    for(int i:sortedSquares(nums))
    {
        cout << i << endl;
    }
}